[Android] Fix context actions icon so that it loads

PureWeen opened the pull request:
### Description of Change ###
- Images loaded from icons don't have a renderer to work from so needed to update the image loading code to work if no renderer is specified
- Fixed menuitem/toolbaritem to correctly set Automation Id using appcompat 

### Breaking changes ###
On Android the AutomationId for Toolbaritems will now set the ContentDescription which makes the UI tests fall more inline with iOS. 

### Issues Resolved ### 
- fixes #6258 


### Platforms Affected ### 
- Android


### Testing Procedure ###
Run the included UI Test

### PR Checklist ###

- [x] Has automated tests <!-- (if tests are omitted or manual, state reason in description) -->
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

paymicro commented on Xamarin.Forms.Platform.Android/Extensions/AccessibilityExtensions.cs:
```diff
@@ -102,20 +103,21 @@ public static string SetNavigationContentDescription(this global::Android.Suppor
-				Control.SetTitle(elemValue);
+				AMenuItemCompat.SetContentDescription(Control, elemValue);
 
 			return;
```
```suggestion
```

paymicro commented on Xamarin.Forms.Platform.Android/ResourceManager.cs:
```diff
@@ -196,10 +195,10 @@ internal static Task ApplyDrawableAsync(this IShellContext shellContext, Bindabl
 
 							// makse sure things are good now that we are back
-							if (element == null || renderer.View == null)
+							if (element == null || (renderer != null && renderer.View == null))
```
In line 176
```C#
if (element == null || renderer?.View == null)
```
I think it's worth doing the same in 2 places.

PureWeen replied:
The reason I didn't do it with the operator is that I don't want a null renderer to return here.

If the renderer is null then this logic doesn't apply. This logic only applies if there is a renderer being passed in

PureWeen replied:
But that does mean I should apply the same thing at line 176 :-p

PureWeen commented on Xamarin.Forms.Platform.Android/Extensions/AccessibilityExtensions.cs:
```diff
@@ -102,20 +103,21 @@ public static string SetNavigationContentDescription(this global::Android.Suppor
-			if (Element == null)
-				return;
+			SetTitleOrContentDescription(Control, (MenuItem)Element);
+		}
```
The main reason I left it in was for the possibilities of an ABI break. I realize that technically I don't need it but it would change the public api so I left it in.

PureWeen commented on Xamarin.Forms.Platform.Android/ResourceManager.cs:
```diff
@@ -174,10 +173,10 @@ internal static Task ApplyDrawableAsync(this IShellContext shellContext, Bindabl
 
 						// makse sure things are good now that we are back
-						if (element == null || renderer.View == null)
+						if (element == null || renderer?.View == null)
```
```suggestion
						if (element == null || (renderer != null && renderer.View == null))
```

hartez commented:
@PureWeen Looks like this is consistently failing the ToolbarItemsTests on all API levels.

hartez commented on Xamarin.Forms.Platform.Android/CellAdapter.cs:
```diff
@@ -334,5 +338,12 @@ bool OnPrepareActionModeImpl(IMenu menu)
 		}
+
+
+		protected override void Dispose(bool disposing)
```
Since this doesn't actually do anything in the Dispose override, do we need it? I realize it's checking `_isDisposed`, but doesn't the `IsDisposed()` extension cover that?

PureWeen replied:
I wasn't sure if it did. IsDisposed() checks the Handle which I wasn't sure if that gets synchronously set to Zero when Dispose is called or not. But I checked with @jonathanpeppers and it looks like just checking if the handle is Zero is the same thing

https://github.com/xamarin/xamarin-android/blob/75e053e5eff431093c7e00d94962fcd13683be5c/src/Mono.Android/Java.Lang/Object.cs#L201-L241

hartez replied:
Yeah, the `IsDisposed` method was just added to make the intent more explicit; otherwise we end up with tons of `Handle == IntPtr.Zero` checks sprinkled about.

PureWeen commented:
@hartez alright I fixed the UI Tests for toolbar items and also added an extra dismiss keyboard call since those same tests are failing elsewhere

I also added @samhouts to review the breaking changes to see if it'll cause accessibility issues

rmarinho commented on Xamarin.Forms.Core.UITests.Shared/Tests/ToolbarItemTests.cs:
```diff
@@ -11,11 +12,11 @@ namespace Xamarin.Forms.Core.UITests
 	[Category(UITestCategories.UwpIgnore)]
```
I think we might disable these tests for now, remove the asserts, take screenshots and mark as manual review

samhouts commented:
@PureWeen I am okay with the breaking change, since this will "fix" `Hint` on ToolbarItems. However, we need to make the behavior consistent across platforms. On iOS and UWP, it will read the `Text`, then the `Hint`. Before this change, it just read the `Text`. With this change, it just reads the `Hint`. When that is fixed, you'll want to update the test. https://github.com/xamarin/Xamarin.Forms/blob/37c22eb1cb64afc5d62b9386a50bb6090446a65a/Xamarin.Forms.Controls/ControlGalleryPages/AutomationPropertiesGallery.cs#L48

PureWeen commented:
@samhouts try it now

PureWeen commented on Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs:
```diff
@@ -46,17 +46,23 @@ internal static void SetAutomationId(AView control, VisualElement element, strin
+			var contentDescription = !string.IsNullOrWhiteSpace(value) ? value : defaultContentDescription;
+
+			if (String.IsNullOrWhiteSpace(contentDescription) && bindableObject is Element element)
+				contentDescription = element.AutomationId;
```
@samhouts is this ok? It doesn't seem like the *ContentDescription* should get overwritten here if the user has set the AutomationId to something

samhouts replied:
works! ship it
